src/reducers/measurements/types.ts

```diff
@@ -0,0 +1,100 @@
+import { measurementIdSymbol } from "../../util/globals";
+
+// -- Measurements JSON types -- //
+
+/**
+ * Measurements are allowed to have arbitrary metadata.
+ * Matching types allowed in Augur's measurements schema
+ * <https://github.com/nextstrain/augur/blob/3f72c40897a80132099729d5d00a6718e76e0e9e/augur/data/schema-measurements.json#L152>
+ */
+type MeasurementMetadata = string | number | boolean
+
+export const measurementsDisplayValues = ["raw", "mean"];
+export type MeasurementsDisplay = (typeof measurementsDisplayValues)[number]
+export const isMeasurementsDisplay = (x: any): x is MeasurementsDisplay => measurementsDisplayValues.includes(x);
+
+interface JsonMeasurement {
+  readonly strain: string
+  readonly value: number
+  readonly [key: string]: MeasurementMetadata
+}
+
+interface JsonCollectionDisplayDefaults {
+  readonly group_by?: string
+  readonly measurements_display?: MeasurementsDisplay
+  readonly show_overall_mean?: boolean
+  readonly show_threshold?: boolean
+}
+
+interface JsonCollectionField {
+  readonly key: string
+  readonly title?: string
+}
+
+interface JsonCollectionGrouping {
+  readonly key: string
+  readonly order?: MeasurementMetadata[]
+}
+
+export interface JsonCollection {
+  readonly display_defaults?: JsonCollectionDisplayDefaults
+  readonly fields?: JsonCollectionField[]
+  readonly filters?: string[]
+  readonly groupings: JsonCollectionGrouping[]
+  readonly key: string
+  readonly measurements: JsonMeasurement[]
+  readonly threshold?: number
+  readonly thresholds?: number[]
+  readonly title?: string
+  readonly x_axis_label: string
+}
+
+export interface MeasurementsJson {
+  readonly collections: JsonCollection[]
+  readonly default_collection?: string
+}
+
+// -- Measurements state types -- //
+
+export interface Measurement {
+  [measurementIdSymbol]: number
+  strain: string
+  value: number
+  [key: string]: string | number
+}
+
+export function isMeasurement(x: any): x is Measurement {
+  return Boolean(x[measurementIdSymbol] !== undefined && x.strain && x.value !== undefined)
+}
+
+export interface Collection {
+  // TODO: Convert this to MeasurementsControlState during parseMeasurementsJSON
+  display_defaults?: JsonCollectionDisplayDefaults
+  fields: Map<string, {title: string}>
+  filters: Map<string, {values: Set<string>}>
+  groupings: Map<string, {values: string[]}>
+  key: string
+  measurements: Measurement[]
+  thresholds?: number[]
+  title?: string
+  x_axis_label: string
+}
+
+export function isCollection(x: any): x is Collection {
+  return Boolean(
+    x.fields &&
+    x.filters &&
+    x.groupings &&
+    x.key &&
+    x.measurements &&
+    x.x_axis_label
+  );
+}
```

Comment:
This is prone to bugs - see https://github.com/nextstrain/auspice/pull/1910#discussion_r1848965620

Reply:
Updated in https://github.com/nextstrain/auspice/pull/1910/commits/0093b4b3addf066682fe2b857840a1c31af3689c